examples/confetti_snapshot_animation/animated_explosion/mod.rs

```diff
@@ -0,0 +1,73 @@
+mod imp;
+
+use crate::{ConfettiWidget, Explosion, ExplosionParameters};
+use glib::clone;
+use glib::subclass::prelude::*;
+use gtk::prelude::*;
+use gtk::{glib, graphene};
+
+glib::wrapper! {
+    pub struct AnimatedExplosion(ObjectSubclass<imp::AnimatedExplosion>);
+}
+
+impl AnimatedExplosion {
+    pub fn new(parameters: ExplosionParameters) -> Self {
+        let this: Self = glib::Object::new(&[]).expect("Failed to create a AnimatedExplosion");
+        let imp = this.imp();
+        imp.explosion.replace(Some(Explosion::new(parameters)));
+        this
+    }
+    pub fn animate(&self, paintable: &ConfettiWidget, duration: u32) {
+        let imp = self.imp();
+
+        // A FrameClock tells the application when to update and repaint a surface.
+        // This may be synced to the vertical refresh rate of the monitor, for example.
+        let frame_clock = paintable.frame_clock().unwrap();
```

Comment:
Why is all the implementation not in the `imp` module? :)

Reply:
Because some of the code requires the implementation, some requires the object instance... I've never fully understood where to put things. Probably I just need to split things in more methods, and put each method in the correct module...

Reply:
You'd pass the object as reference forward usually, see the shape of the methods in the Impl traits.